Build/init

Thienhuynh95 opened the pull request:
## Description
<!-- Explain what this Pull Request should do -->
<!-- Example:
This PR fixes a bug in our elastic search order query.
-->
This PR publishes the first version of zonos-typescript-sdk.
- added github workflows
- added build process for the sdk
- added codegen for generating schema and types
- added test to make sure response and error types format are correct

Thienhuynh95 committed:
build: init project

Thienhuynh95 committed:
build: first version for the sdk
- added github workflows
- added build process for the sdk
- added codegen for generating schema and types
- added test to make sure response and error types format are correct

Thienhuynh95 committed:
build: update node version and command

Thienhuynh95 committed:
build: add env to ci action

goodpickle commented on package.json:
```diff
@@ -0,0 +1,89 @@
+    "graphql-request": "^4.3.0",
+    "graphql-tag": "^2.12.6"
+  },
+  "engines": {
+    "node": "18"
+  },
```
What do you think about removing the consuming projects to a particular node version?
```suggestion
```

Thienhuynh95 replied:
This's removed in the packed version, so the consuming project will not be limited to the node version. I just restrict it for whoever developing this project. Below is what consuming project will get.
![image](https://user-images.githubusercontent.com/17950626/229246054-d21fb27b-2b9a-4f5f-a589-f51d9aa665e5.png)

evad1n commented on package.json:
```diff
@@ -0,0 +1,89 @@
+    "graphql-request": "^4.3.0",
+    "graphql-tag": "^2.12.6"
+  },
+  "engines": {
+    "node": "18"
+  },
```
Considering our plugins user various lower versions of node (I believe), I wonder if maybe we should expand this range?

Thienhuynh95 replied:
Yah, that's why I remove the engine when preparing for publishing to npm

Thienhuynh95 replied:
The reason for this node version on package.json is just for unit testing to use built-in `fetch` of node 18

evad1n commented on pull_request_template.md:
```diff
@@ -0,0 +1,11 @@
+## Issue
+<!-- Example:
+Error when importing the sdk
+-->
+
+## Description
+<!-- Explain what this Pull Request should do -->
+<!-- Example:
+This PR fixes a bug in our elastic search order query.
+-->
+This PR
```
What do you think about moving this file inside the `.github` folder, as it organizes better in my opinion.

Thienhuynh95 replied:
Thanks for the suggestion.

evad1n commented on src/gqlRequest.ts:
```diff
@@ -0,0 +1,120 @@
+  Sdk as ZonosCustomerGraphSdk,
+} from 'src/types/generated/graphql.zonos-customer-graph.types';
+
+type Schema = 'zonos-customer-graph' | 'auth';
```
What is the reason for the `zonos` prefix on one and not the other?

Thienhuynh95 replied:
😆 Good catch. Thanks

goodpickle commented on codegen/auth.codegen.ts:
```diff
@@ -0,0 +1,27 @@
+import type { TypeScriptDocumentsPluginConfig } from '@graphql-codegen/typescript-operations';
```
What do you think about using the graph prefix `zonos-auth-graph` like in `documentation`?

Thienhuynh95 replied:
Sounds good. Thanks for the suggestion.

evad1n commented on .github/actions/auto-assign/package.json:
```diff
@@ -0,0 +1,23 @@
+    "@octokit/webhooks-types": "6.9.0",
+    "@types/node": "^18.11.0",
+    "@vercel/ncc": "0.34.0",
+    "typescript": "^4.9.4"
```
There is a newer version of typescript, what do you think about using the latest for this new package?

evad1n replied:
haha whoops this is the wrong package.json, still valid technically but you can ignore.

Thienhuynh95 replied:
It's valid. Upgraded. Thanks for catching that.

evad1n commented on package.json:
```diff
@@ -0,0 +1,89 @@
+    "ts-node": "^10.9.1",
+    "ts-toolbelt": "^9.6.0",
+    "tsconfig-paths-webpack-plugin": "^4.0.1",
+    "ttypescript": "^1.5.15",
+    "typescript": "^4.9.5",
```
Hmmm, there is a newer version of typescript now, but I know from experience that the `ttypescript` doesn't work yet with it. I wonder if we can find another option. Some quick digging: https://github.com/cevek/ttypescript/issues/140#issuecomment-1488575848

Thienhuynh95 replied:
Yah, at first I upgraded to typescript 5, but I downgraded back to 4.9.5. And that's the reason why 😢 
I'm not sure what will be the alternative yet, but I will do more research on it.

evad1n replied:
Even if there isn't anything ready yet. I think it's worth keeping up to date to keep up with newer versions of typescript.

Thienhuynh95 replied:
Yes, I agree, but there isn't a version of `ttypescript` or `ts-patch` that works with typescript 5 yet. I'll try to upgrade when there is one. Thanks

Thienhuynh95 committed:
refactor: upgrade typescript for auto-assign

Thienhuynh95 committed:
refactor: organize github file

Thienhuynh95 committed:
refactor: update build script

Thienhuynh95 committed:
build: remove auth graph
- because this graph need additional service token to perform an action,
and that token is only available to internal of zonos.

goodpickle committed:
feat(npm): release package

Thienhuynh95 committed:
refactor: update test for currencyFormat query

Thienhuynh95 committed:
Merge branch 'build/init' of github.com:Zonos/zonos-typescript-sdk into build/npm-release-package

Thienhuynh95 committed:
build: update release action

goodpickle committed:
Merge pull request #2 from Zonos/build/npm-release-package

feat(npm): release package
